src/Mixins/Expectation.php

```diff
@@ -560,6 +560,10 @@ public function toBeNull(string $message = ''): self
      */
     public function toHaveKey(string|int $key, mixed $value = new Any(), string $message = ''): self
     {
+        if (! is_iterable($this->value)) {
```

Comment:
I was guided by the idea to ensure the Expectation value is an `iterable` to avoid this behavior present in v1:

```php 
test('using string')->expect('long string to have key')->toHaveKey(0); //Passes
``` 

However, this also passes on v1 and checking for iterable would break it:

```php
class ExampleClass
{
    public $foo = 'foo';
}

test('pass with a class')->expect((new ExampleClass()))->toHaveKey('foo');

//is_iterable(new stdClass()) === false
```

I decided to avoid a breaking change in favor of user experience in v2.

So, I came up with some changes but I'm feeling a bit conflicted and not quite happy with the code.